plugins/scaffolder-backend/src/scaffolder/tasks/DatabaseTaskStore.ts

```diff
@@ -462,4 +479,52 @@ export class DatabaseTaskStore implements TaskStore {
       body: serializedBody,
     });
   }
+
+  async recoverTasks(options: TaskStoreRecoverTaskOptions): Promise<string[]> {
+    const taskIdsToRecover: string[] = [];
+    const timeoutS = Duration.fromObject(options.timeout).as('seconds');
+
+    await this.db.transaction(async tx => {
+      let heartbeatInterval = this.db.raw(
+        `? - interval '${timeoutS} seconds'`,
+        [this.db.fn.now()],
+      );
+      if (this.db.client.config.client.includes('mysql')) {
+        heartbeatInterval = this.db.raw(
+          `date_sub(now(), interval ${timeoutS} second)`,
+        );
+      } else if (this.db.client.config.client.includes('sqlite3')) {
+        heartbeatInterval = this.db.raw(`datetime('now', ?)`, [
+          `-${timeoutS} seconds`,
+        ]);
+      }
```

Comment:
Break this out of this transaction for readability sake. The catalog also has some examples.

//freben